src/capabilities/execute-dql.ts

```diff
@@ -13,15 +13,29 @@ export const verifyDqlStatement = async (dtClient: HttpClient, dqlStatement: str
   return response;
 };
 
+/**
+ * Execute a DQL statement against the Dynatrace API.
+ * If the result is immediately available, it will be returned.
+ * If the result is not immediately available, it will poll for the result until it is available.
+ * @param dtClient
+ * @param dqlStatement - The DQL statement to execute.
+ * @param maxResultRecords - Optional: The maximum number of result records to return.
+ * @param maxResultBytes - Optional: The maximum number of result bytes to return.
+ * @returns the result without metadata and without notifications, or undefined if the query failed or no result was returned.
+ */
 export const executeDql = async (
   dtClient: HttpClient,
   dqlStatement: string,
+  maxResultRecords?: number,
+  maxResultBytes?: number,
```

Comment:
If we expect that this could grow with more options (e.g., sampling), it might make sense to use an options object. This could be especially useful if some invocations use "later" arguments and don't care about former ones, so that the call doesn't have to look like `executeDql(client, 'fetch logs...', undefined, undefined, undefined, undefined, 42)`

It's not a biggie now, but it might be easier to use options objects now to avoid changing the signature later on.